eventsourcing/src/test/java/org/axonframework/eventsourcing/eventstore/AbstractEventStorageEngineTest.java

```diff
@@ -44,12 +45,21 @@
 
     private AbstractEventStorageEngine testSubject;
 
+    @DirtiesContext
+    @Test
+    public void testUniqueKeyConstraintOnFirstEventIdentifier() {
```

Comment:
Darn, of course, an aggregate's  uniqueness is guaranteed through the aggregate id and sequence number, meaning that the aggregate type does not matter at all. Hence, the exception message (which I suggested...) which adds the aggregate type is incorrect since you cannot be certain the event you are inserting is for the same aggregate type. 

I'll suggest a different message in the previous file, sorry for the inconvenience.

Reply:
This also means the exception should be named slightly different, as it's not the aggregate which already exists, it's the aggregate identifier.